mantidproject/mantid
# Refactor PeakShape

Description of work.

This refactors `Mantid::Geometry::PeakShape::radius()` to better support arbitrary shapes without casting back to it's derived type. I also made internal changes to `vtkPeaksFilter` to remove many unnecessary copies.

**To test:**



Open a `MDEventWorkspace` and a `PeaksWorkspace` in the VSI. View the peaks table to activate the PeaksFilter and verify that the points and glyphs still render properly.

This pull request has no related issue.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
